File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/human-resources/employee-address/employee-address-lookup-model.ts

export interface IEmployeeAddressLookupModel {
  employeeID: number;
  addressID: number;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class EmployeeAddressLookupModel implements IEmployeeAddressLookupModel {
  employeeID: number;
  addressID: number;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<EmployeeAddressLookupModel>) {
    if (!!init) {
      Object.assign<EmployeeAddressLookupModel, Partial<EmployeeAddressLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/HumanResources/EmployeeAddress/Commands/Queries/GetEmployeeAddresses/EmployeeAddressLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.HumanResources;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Person.Address.Queries.GetAddresses;
namespace AdventureWorks.Application.DataEngine.HumanResources.EmployeeAddress.Queries.GetEmployeeAddresses
{
    public partial class EmployeeAddressLookupModel : IHaveCustomMapping
    {
        public int EmployeeID { get; set; }
        public int AddressID { get; set; }
        public Guid Rowguid { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.HumanResources.EmployeeAddress, EmployeeAddressLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<EmployeeAddressLookupModel, BaseEmployeeAddress>().IgnoreMissingDestinationMembers();
        }
    }
}
